src/Jackett.Common/Utils/StringUtil.cs

```diff
@@ -138,13 +138,35 @@ public static string MakeValidFileName(string text, char? replacement = '_', boo
             return changed ? sb.ToString() : text;
         }
 
-        public static string GetQueryString(this NameValueCollection collection, Encoding encoding = null) =>
-            string.Join("&", collection.AllKeys.Select(a =>
-                $"{a}={WebUtilityHelpers.UrlEncode(collection[a], encoding ?? Encoding.UTF8)}"));
+        /// <summary>
+        /// Converts a NameValueCollection to an appropriately formatted query string.
+        /// Duplicate keys are allowed in a NameValueCollection, but are stored as a csv string in Value.
+        /// This function handles leaving the values together in the csv string or splitting the value into separate keys
+        /// </summary>
+        /// <param name="collection">The NameValueCollection being converted</param>
+        /// <param name="encoding">The Encoding to use in url encoding Value</param>
+        /// <param name="splitMultiValues">Whether the Value should be Key=Value or split into Key=Val1&Key=Val2</param>
+        /// <param name="separator">The string used to separate each query value</param>
+        /// <returns>A web encoded string of key=value parameters separated by the separator</returns>
+        public static string GetQueryString(this NameValueCollection collection, Encoding encoding = null,
+                                            bool splitMultiValues = false, string separator = "&")
```

Comment:
We are changing the default behavior, this is ok but I think you should take a look at the current usages..

Reply:
The default behavior remains the same as it was. With `splitMultiValues == false` and `separator == "&"` the code that runs is:
```
return string.Join(separator/*=="&"*/, collection.AllKeys.Select(
                    a => $"{a}={WebUtilityHelpers.UrlEncode(collection[a], encoding ?? Encoding.UTF8)}"));
```
which is the original setup of `Key=Val1,Val2` and the reason why you couldn't use NameValueCollection in your PR.